tests/dialects/test_redshift.py

```diff
@@ -226,7 +226,6 @@ def test_identity(self):
         self.validate_identity("SELECT * FROM #x")
         self.validate_identity("SELECT INTERVAL '5 day'")
         self.validate_identity("foo$")
-        self.validate_identity("$foo")
```

Comment:
This is invalid Redshift
<img width="346" alt="Screenshot 2023-09-26 at 8 48 57 PM" src="https://github.com/tobymao/sqlglot/assets/46752250/4dc89f08-453a-4c77-bbaa-05320edfd101">